src/Compatibility/Microsoft.DotNet.PackageValidation/Package.cs

```diff
@@ -19,126 +20,137 @@ namespace Microsoft.DotNet.PackageValidation
         /// </summary>
-        public IEnumerable<ContentItem> RuntimeSpecificAssets { get; private set; }
+        public IEnumerable<ContentItem> RuntimeSpecificAssets { get; }
 
         /// <summary>
         /// List of all the runtime assets in the package.
         /// </summary>
-        public IEnumerable<ContentItem> RuntimeAssets { get; private set; }
+        public IEnumerable<ContentItem> RuntimeAssets { get; }
 
         /// <summary>
-        /// Checks if there are any assemblies under ref folder in the package.
+        /// List of rids in the package.
         /// </summary>
-        public bool HasRefAssemblies => RefAssets.Any();
+        public IEnumerable<string> Rids { get; }
 
         /// <summary>
-        /// List of rids in the package.
+        /// List of the frameworks in the package.
         /// </summary>
-        public IEnumerable<string> Rids { get; private set; }
+        public IEnumerable<NuGetFramework> FrameworksInPackage { get; }
+
+        internal Package(string packagePath, string packageId, string version, IEnumerable<string> packageAssets, Dictionary<NuGetFramework, IEnumerable<PackageDependency>> packageDependencies, RuntimeGraph? runtimeGraph)
+        {
+            PackagePath = packagePath;
+            PackageId = packageId;
+            Version = version;
+            PackageDependencies = packageDependencies;
+
+            _packageAssets.Load(packageAssets);
+            _conventions = new ManagedCodeConventions(runtimeGraph);
+
+            PackageAssets = _packageAssets.FindItems(_conventions.Patterns.AnyTargettedFile);
+            RefAssets = _packageAssets.FindItems(_conventions.Patterns.CompileRefAssemblies);
+            LibAssets = _packageAssets.FindItems(_conventions.Patterns.CompileLibAssemblies);
+            CompileAssets = RefAssets.Any() ? RefAssets : LibAssets;
+            RuntimeAssets = _packageAssets.FindItems(_conventions.Patterns.RuntimeAssemblies);
+            RuntimeSpecificAssets = RuntimeAssets.Where(t => t.Path.StartsWith("runtimes")).ToArray();
+            Rids = RuntimeSpecificAssets.Select(t => (string)t.Properties["rid"]).ToArray();
+            FrameworksInPackage = CompileAssets.Select(t => (NuGetFramework)t.Properties["tfm"])
+                .Concat(RuntimeAssets.Select(t => (NuGetFramework)t.Properties["tfm"]))
+                .Distinct()
+                .ToArray();
+        }
 
         /// <summary>
-        /// List of the frameworks in the package.
+        /// Creates the package object from package path.
         /// </summary>
-        public IEnumerable<NuGetFramework> FrameworksInPackage { get; private set; }
+        /// <param name="packagePath">The path to the package path.</param>
+        /// <param name="runtimeGraph">The path to the the runtime graph.</param>
+        /// <returns>The package object.</returns>
+        public static Package Create(string? packagePath, string? runtimeGraph)
```

Comment:
Is the code here a direct copy-paste from what we had in NupkgParser? If so, any reason why we moved it? If we just want it to be a static member on the same class we could have just kept the code separate and make this package type partial.

Reply:
It is usually good to keep files as a boundary for logic, it makes the code easier to find and more readable.

Reply:
Absolutely agree with keeping files as a boundary for logic. That's why I moved the Create method into Package as the creation of the package belongs into the Package itself. The Package type already heavily relies on NuGet apis, there's no separate that.

> Is the code here a direct copy-paste from what we had in NupkgParser?

Not exactly. It merges the previously existing methods together and leverages the compiler's nullability feature now: https://github.com/dotnet/sdk/blob/4fdc385bc22e9237fef3301777ed71dc4dcff714/src/Compatibility/Microsoft.DotNet.PackageValidation/NupkgParser.cs#L24-L48.